addons/crm_partner_assign/wizard/crm_channel_interested.py

```diff
@@ -1,59 +1,50 @@
+        self.ensure_one()
+        if self.interested and not self.contacted:
             raise UserError(_("You must contact the lead before saying that you are interested"))
-        lead_obj = self.pool.get('crm.lead')
-        lead_obj.check_access_rights(cr, uid, 'write')
-        if wizard.interested:
+        CrmLead = self.env['crm.lead']
+        CrmLead.check_access_rights('write')
+        leads = self.env.context.get('active_ids', [])
+        if self.interested:
             message = _('<p>I am interested by this lead.</p>')
             values = {}
         else:
             stage = 'stage_portal_lead_recycle'
-            if wizard.contacted:
+            if self.contacted:
                 message = _('<p>I am not interested by this lead. I contacted the lead.</p>')
             else:
                 message = _('<p>I am not interested by this lead. I have not contacted the lead.</p>')
             values = {'partner_assigned_id': False}
-            user = self.pool.get('res.users').browse(cr, uid, uid, context=context)
-            partner_ids = self.pool.get('res.partner').search(cr, SUPERUSER_ID, [('id', 'child_of', user.partner_id.commercial_partner_id.id)], context=context)
-            lead_obj.message_unsubscribe(cr, SUPERUSER_ID, context.get('active_ids', []), partner_ids, context=None)
-            try:
-                stage_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'crm_partner_assign', stage)[1]
-            except ValueError:
-                stage_id = False
+            partners = self.env['res.partner'].sudo().search([('id', 'child_of', self.env.user.partner_id.commercial_partner_id.id)])
+            CrmLead.sudo().browse(leads).message_unsubscribe(partners.ids)
+            stage_id = self.env.ref("crm_partner_assign.%s" % (stage)).id
             if stage_id:
                 values.update({'stage_id': stage_id})
-        if wizard.comment:
-            message += '<p>%s</p>' % wizard.comment
-        for active_id in context.get('active_ids', []):
-            lead_obj.message_post(cr, uid, active_id, body=message, subtype="mail.mt_comment", context=context)
+        if self.comment:
+            message += '<p>%s</p>' % self.comment
+        for active_id in leads:
+            CrmLead.browse(active_id).message_post(body=message, subtype="mail.mt_comment")
         if values:
-            lead_obj.write(cr, SUPERUSER_ID, context.get('active_ids', []), values)
-        if wizard.interested:
-            for lead in lead_obj.browse(cr, uid, context.get('active_ids', []), context=context):
-                lead_obj.convert_opportunity(cr, SUPERUSER_ID, [lead.id], lead.partner_id and lead.partner_id.id or None, context=None)
+            CrmLead.sudo().browse(leads).write(values)
+        if self.interested:
+            for lead in CrmLead.browse(leads):
+                CrmLead.sudo().browse(lead.id).convert_opportunity(lead.partner_id and lead.partner_id.id or None)
```

Comment:
Why not browse leads at top and use recordset everywhere instead of browsing each time